plugin/trino-druid/src/main/java/io/trino/plugin/druid/aggregate/SingleInputAggregateFunction.java

```diff
@@ -0,0 +1,132 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.druid.aggregate;
+
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcExpression;
+import io.trino.plugin.jdbc.JdbcTypeHandle;
+import io.trino.plugin.jdbc.expression.AggregateFunctionRule;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.Type;
+
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verifyNotNull;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.expressionType;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.outputType;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.variable;
+import static java.util.Objects.requireNonNull;
+
+//TODO Move this class to base-jdbc module and update all jdbc connectors to leverage it.
+public class SingleInputAggregateFunction
+        implements AggregateFunctionRule
+{
+    private static final Capture<Variable> INPUT = newCapture();
+    private final String prestoName; // presto aggregate function name to match
+    private Type inputType; // empty if all input types should match, or the set to match
+    private Type outputType; // provide if the pattern should only match specific output type
+    private final Optional<String> expression; // the expression format
+    private final Optional<JdbcTypeHandle> jdbcTypeHandle; // type handle if its different from input column
+
+    public SingleInputAggregateFunction(
+            String prestoName,
+            Type inputType, Type outputType, Optional<String> expression,
+            Optional<JdbcTypeHandle> jdbcTypeHandle)
+    {
+        this.prestoName = requireNonNull(prestoName, "prestoName is null");
+        this.inputType = requireNonNull(inputType, "inputType is null");
+        this.outputType = requireNonNull(outputType, "outputType is null");
+        this.expression = requireNonNull(expression, "expression is null");
+        this.jdbcTypeHandle = requireNonNull(jdbcTypeHandle, "jdbcTypeHandle is null");
+    }
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().equalTo(prestoName))
+                .with(outputType().equalTo(outputType))
+                .with(singleInput()
+                        .matching(variable().with(expressionType().equalTo(inputType)).capturedAs(INPUT)));
```

Comment:
Some functions can work for multiple input types too? Is the suggested usage to add a separate rule for each type?

Just wanted to get clarification - no changes requested yet.